project/stac_api/models.py

```diff
@@ -1,28 +1,307 @@
+    # proterties: datetime, eo_gsd and title (the rest is hence commented out)
+    properties_datetime = models.DateTimeField()
+    # properties_eo_bands = model.TextFields(blank=True)  # ? [string]?
+    # properties_eo_cloud_cover = models.FloatField(blank=True)  # float
+    properties_eo_gsd = models.FloatField(blank=True)
+    # properties_instruments = models.TextField(blank=True)  # [string]
+    # properties_license = models.TextField(blank=True)  # string
+    # properties_platform = models.TextField(blank=True)  # string
+    # properties_providers is implemented as separate class "Provider_Item" with
+    # many-to-many relations to class Item objects
+    properties_title = models.CharField(blank=True)  # string
+    # properties_view_off_nadir = models.FloatField(blank=True)
+    # properties_view_sun_azimuth = models.FloatField(blank=True)
+    # properties_view_elevation = models.FloatField(blank=True)
+
+    # after discussion with Chris and Tobias:
+    # stac_extension will be populated with default values that are set to be
+    # non-editable for the moment. Could be changed, should the need arise.
+    stac_extension = ArrayField(
+        models.CharField(),
+        default=[
+            'eo',
+            'proj',
+            'view',
+            'https://data.geo.admin.ch/stac/geoadmin-extension/1.0/schema.json'
+        ],
+        editable=False
+    )
+
+    stac_version = models.CharField(blank=False)
+    GeoJSON_type = models.CharField(default="Feature", blank=False, editable=False)
+    # specifies the GeoJSON type and MUST be set to "Feature".
+    assets = models.TextField()
+    # this is defined as required here: https://github.com/radiantearth/stac-spec/blob/v0.9.0/item-spec/item-spec.md
+    # and is meant to contain a dictionary of asset objects than can be downloaded, each with a unique key
+    # will be auto-populated on every update of an asset inside this item.
+    # TODO: overwrite assets save() function accordingly.
+    location = models.URLField()
+
+    def __str__(self):
+        return self.item_name
+
+    def clean(self):
+        if len(self.SW) != len(self.NE):
+            raise ValidationError(_(
+                'If intended spatial extent is 3D, then elevation needs to be defined in southwesterly most and northeasterly most point.'
+            ))
+        elif (len(self.SW) != 4) and (len(self.SW) != 6):
+            raise ValidationError(_('Bounding box incorrectly defined.'))
+
+    def save(self, *args, **kwargs):
+        #TODO: on every item update:
+        # 1. re-calculate the envelope over all bboxes of all items inside this collection
+        #    and update the collections' bbox accordingly
+        #
+        # 2. loop over all items inside the collection and set the "start_date" of
+        #    the collection to the oldest date found in item dates. Same for
+        #    collections' end_date (set to most recent date found in items)
```

Comment:
Not sure it's necessary to always loop over all items (this will be a performance nightmare): Imho it's sufficient to just compare the current item with the values stored on the collection level, and adapt if necessary (only thing to consider is to do proper row locking on the collection object (or use `update()`)).

Reply:
That's correct. Looping over all items would be very cost intensive and does not make sense. It's enough to only check the currently added item. I will adapt it this way.